### ISSUE TITLE
gcs : The request signature we calculated does not match the signature you provided

### ISSUE THREAD

**Issue by spff** [ClickHouse/ClickHouse#37939]
Similar to https://github.com/ClickHouse/ClickHouse/issues/33009 & https://github.com/ClickHouse/ClickHouse/pull/37344, but I not using s3 but url (I signed it via python and passed to clickhouse), can we perform the same action to all the request's URI passed to `sendRequest` as https://github.com/ClickHouse/ClickHouse/pull/37344 did?

```
src/Storages/MeiliSearch/MeiliSearchConnection.cpp:40:    std::ostream & os = session.sendRequest(req);
src/Storages/MeiliSearch/MeiliSearchConnection.cpp:75:    session.sendRequest(req);
src/IO/WriteBufferFromHTTP.cpp:33:    ostr = &session->sendRequest(request);
src/IO/S3/PocoHTTPClient.cpp:253:            auto & request_body_stream = session->sendRequest(poco_request);
src/IO/ReadWriteBufferFromHTTP.h:182:                auto & stream_out = sess->sendRequest(request);
src/Processors/Formats/Impl/AvroRowInputFormat.cpp:680:                    session->sendRequest(request);
src/Processors/Transforms/MongoDBSource.cpp:58:        connection.sendRequest(*command, response);
src/Processors/Transforms/MongoDBSource.cpp:113:        connection.sendRequest(*command, response);
src/Disks/ObjectStorages/S3/ProxyResolverConfiguration.cpp:70:                session->sendRequest(request);
```

Another question: Would you recommend using `url` or `s3` for downloading files from s3/gcs in regard to performance? I found they will both perform ranged download, will `s3` first do some bucket list? I also found that `s3` supports `compression` while the `url` doesn't (although it seems that `URL` table engine does support)

**Comment by filimonov** [comment]
Sorry, it's very hard to follow your explanations. 

How `url` tables function relates to gcs? 

What exactly behavior is unexpected for you? Can you show the example of wrong behavior and also explain the desired one? It will help to create a test case and fix it.

**Comment by alexey-milovidov** [comment]
It's very likely that the issue is fixed because ClickHouse started to support GCS officially: https://clickhouse.com/blog/clickhouse-cloud-on-google-cloud-platform-gcp-is-generally-available

**Comment by spff** [comment]
Not fixed.

**Comment by alexey-milovidov** [comment]
Try to pass `=` as is instead of encoding it as `%3D`:

```
select * from url('https://storage.googleapis.com/ch-test/bad=/a.parquet?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=sys-da-dev%40appier-data-analytics.iam.gserviceaccount.com%2F20220616%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20220616T005704Z&X-Goog-Expires=604800&X-Goog-SignedHeaders=host&X-Goog-Signature=8c5970ec0fcc9f2da2262ea1a2ee785fee5f76fcf30113400debaa1fb52b30f2fbf6f1736c12a68c7e07054c03b8e6930275f3d5f7af703c6d43755fa33180dffcf8512d5ea4b68c41bca5061ab960c93de24c15c89679319594749196742630268d17daed36711ad2a262021076abcfe0f4a1f323b05bd5cfd7d984a8dea7f9e9384b69dad2c83254ece3c5299fb8c7fb440bfba7a7bb2fc948787a61bfb5a97088f1be61c0aff985f7f6c595bd0cecd2a54d3729e53c81fa8d85c77eb845996723f6ebcdfc8e6a2aa8cbce736a86c05e340496266d12a9b31762f8918c9cf03e836312d6bc34ebec69bd346d089bf6192d5fac65a61090d038fcc1eef58c3e', 'Parquet')

```

**Comment by spff** [comment]
Doesn't work, same err (SignatureDoesNotMatch).
